src/docs/rust.mdx

```diff
@@ -0,0 +1,324 @@
+---
+title: Rust Development at Sentry
+---
+
+This is a document that contains a bunch of useful resources for getting started with Rust and adhering to our coding principles.
+
+## Getting Started
+
+- Join the `#discuss-rust` channel in Slack
+- A quick introduction into the Syntax for first-timers:
+[https://fasterthanli.me/blog/2020/a-half-hour-to-learn-rust/](https://fasterthanli.me/blog/2020/a-half-hour-to-learn-rust/)
+- The Rust Book, comprehensively documenting the language:
+[https://doc.rust-lang.org/book/](https://doc.rust-lang.org/book/)
+- We follow the Rust style (enforced by `rustfmt` and `clippy` where possible):
+[https://doc.rust-lang.org/1.0.0/style/README.html](https://doc.rust-lang.org/1.0.0/style/README.html)
+- The Async Book:
+[https://rust-lang.github.io/async-book/](https://rust-lang.github.io/async-book/)
+
+## Advanced Reading
+
+- The Little Book of Rust Macros
+[https://danielkeep.github.io/tlborm/book/index.html](https://danielkeep.github.io/tlborm/book/index.html)
+- Rust Nomicon: The Dark Arts of Unsafe Rust
+[https://doc.rust-lang.org/nomicon/](https://doc.rust-lang.org/nomicon/)
+- Rust by Example: A comprehensive list of examples
+[https://doc.rust-lang.org/stable/rust-by-example/](https://doc.rust-lang.org/stable/rust-by-example/)
+- [https://doc.rust-lang.org/1.15.1/book/choosing-your-guarantees.html](https://doc.rust-lang.org/1.15.1/book/choosing-your-guarantees.html)
+
+
+## Coding Principles
+
+
+### Import order
+
+Imports must be declared before any other code in the file, and can only be preceded by module doc comments and module attributes (`#![...]`). We bundle imports in groups based on their origin, each separated by an empty line. The order of imports is:
+
+1. Rust standard library, including `std`, `core` and `alloc`
+2. External dependencies
+3. Workspace dependencies
+4. Crate modules
```

Comment:
@jan-auer you recently suggested relaxing these constrictions. Instead of explicitly listing rules, should we just refer to what autoformatter we should use?